rebuild/src/secret_key.cc

```diff
@@ -0,0 +1,171 @@
+        std::ostringstream msg;
+        msg << "ikm must be greater than or equal to "
+            << BLST_TS_SECRET_KEY_LENGTH << " bytes";
+        Napi::TypeError::New(env, msg.str()).ThrowAsJavaScriptException();
+        return env.Undefined();
+    }
+
+    bool is_external = true;
+    Napi::Object wrapped = module->_secret_key_ctr.New(
+        {Napi::External<bool>::New(env, &is_external)});
+    SecretKey *sk = SecretKey::Unwrap(wrapped);
+
+    // If `info` string is passed use it otherwise use default without
+    if (!info[1].IsUndefined()) {
+        if (!info[1].IsString()) {
+            Napi::TypeError::New(env, "BLST_ERROR: info must be a string")
+                .ThrowAsJavaScriptException();
+            return env.Undefined();
+        }
+        sk->_key->keygen(
+            ikm.Data(),
+            ikm.ByteLength(),
+            info[1].As<Napi::String>().Utf8Value());
+    } else {
+        sk->_key->keygen(ikm.Data(), ikm.ByteLength());
+    }
+
+    // Check if key is zero and set flag if so. Several specs depend on this
+    // check (signing with zero key). Do after building instead of checking
+    // incoming bytes incase info changes the key
+    blst::byte key_bytes[BLST_TS_SECRET_KEY_LENGTH];
+    sk->_key->to_bendian(key_bytes);
+    if (is_zero_bytes(key_bytes, 0, BLST_TS_SECRET_KEY_LENGTH)) {
+        sk->_is_zero_key = true;
+    }
+
+    return scope.Escape(wrapped);
+}
+
+Napi::Value SecretKey::Deserialize(const Napi::CallbackInfo &info) {
+    BLST_TS_FUNCTION_PREAMBLE(info, env, module)
+
+    Napi::Value sk_bytes_value = info[0];
+    BLST_TS_UNWRAP_UINT_8_ARRAY(sk_bytes_value, sk_bytes, "skBytes")
+    std::string err_out{"BLST_ERROR: skBytes"};
+    if (!is_valid_length(
+            err_out, sk_bytes.ByteLength(), BLST_TS_SECRET_KEY_LENGTH)) {
+        Napi::TypeError::New(env, err_out).ThrowAsJavaScriptException();
+        return env.Undefined();
+    }
+
+    bool is_external = true;
+    Napi::Object wrapped = module->_secret_key_ctr.New(
+        {Napi::External<bool>::New(env, &is_external)});
+    SecretKey *sk = SecretKey::Unwrap(wrapped);
+
+    // Check if key is zero and set flag if so. Several specs depend on this
+    // check (signing with zero key)
+    if (is_zero_bytes(sk_bytes.Data(), 0, sk_bytes.ByteLength())) {
+        sk->_is_zero_key = true;
+    }
+    // Deserialize key
+    sk->_key->from_bendian(sk_bytes.Data());
+
+    return scope.Escape(wrapped);
+}
+
+SecretKey::SecretKey(const Napi::CallbackInfo &info)
+    : Napi::ObjectWrap<SecretKey>{info},
+      _key{std::make_unique<blst::SecretKey>()},
+      _is_zero_key{false} {
+    Napi::Env env = info.Env();
+    Napi::HandleScope scope(env);
+    // Check that constructor was called from C++ and not JS. Externals can only
+    // be created natively.
+    if (!info[0].IsExternal()) {
+        Napi::Error::New(env, "SecretKey constructor is private")
+            .ThrowAsJavaScriptException();
+        return;
```

Comment:
What's the point of this `return`?

Reply:
copy/pasta leftover.  fixed in 2e11c9e548e8aad4c21354cedd4416baff8652ef